code-monkey/src/instrumentation/__init__.py

```diff
@@ -64,11 +72,41 @@ def initialize_tracer(attributes: Attributes = None):
     set_tracer(trace.get_tracer("replayio.ai-playground"))
 
 
-def instrument(name: str, attributes=None):
+def instrument(name: str, params: List[str] = None, attributes: Attributes = None):
     def decorator(func):
         @functools.wraps(func)
         def wrapper(*args, **kwargs):
-            with tracer().start_as_current_span(name, attributes=attributes):
+            span_attributes = attributes.copy() if attributes else {}
+            
+            if params:
```

Comment:
This also looks suspiciously AI-y :smile: 

(My point being that *sometimes* its Ok to go with less-than-readable code, if it gets the job done :eyes: )

Reply:
claude and I wrote that together